pkg/controller/workload/job/interface.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package job
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+)
+
+type GenericJob interface {
+	// Object returns the job instance.
+	Object() client.Object
+	// IsSuspend returns whether the job is suspended or not.
+	IsSuspend() bool
+	// Suspend will suspend the job.
+	Suspend() error
+	// UnSuspend will unsuspend the job.
+	UnSuspend() error
```

Comment:
maybe just `Suspend(s bool) error`

Reply:
I actually prefer to keep Suspend / UnSuspend by analogy to startJob, stopJob.